hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodieClientMultiWriter.java

```diff
@@ -136,6 +138,62 @@ public void testMultiWriterWithAsyncTableServicesWithConflictMOR() throws Except
     testMultiWriterWithAsyncTableServicesWithConflict(HoodieTableType.MERGE_ON_READ);
   }
 
+  @ParameterizedTest
+  @EnumSource(value = HoodieTableType.class, names = {"COPY_ON_WRITE", "MERGE_ON_READ"})
```

Comment:
Can you confirm something, is this test failing in master and succeeding with this patch? 
I feel this is not testing the actual fix. 
Let me try to explain whats the scenario we need to test. 

c1: succeeds
c2: partially failed commit. 

after expiration: 

c3 and c4: concurrently starts. 
and cleaning kicks in by one of them let's say c3. During this, it will attempt to rollback any partially failed writes. And due to the bug, it will delete heartbeats for both C2 and C4 as well. With the fix in this patch, only c2 heartbeat should be deleted.

Reply:
unless cleaning kicks in, rollback of failed writes may not get triggered.